src/elements/element.point.js

```diff
@@ -79,7 +79,7 @@ module.exports = Element.extend({
 
 		// Clipping for Points.
 		if (chartArea === undefined || (model.x >= chartArea.left && chartArea.right * errMargin >= model.x && model.y >= chartArea.top && chartArea.bottom * errMargin >= model.y)) {
-			ctx.strokeStyle = vm.borderColor || defaultColor;
+			ctx.strokeStyle = ctx.lineWidth === 0 ? 'rgba(0,0,0,0)' : vm.borderColor || defaultColor;
```

Comment:
You are trying to access `ctx.lineWidth` **before** assigning it a value (next line), is that expected?

Though, I think [@benmccann suggestion](https://github.com/chartjs/Chart.js/pull/5180#discussion_r200826310) was wrong because [according the MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineWidth):

> lineWidth: A number specifying the line width in space units. **Zero, negative, Infinity and NaN values are ignored.**

![image](https://user-images.githubusercontent.com/3874900/45594818-3d0dfc00-b9a1-11e8-91e7-2a199b9b9e86.png)

So `ctx.lineWidth` will never match `ctx.lineWidth === 0` (because ignored) so I would keep your first approach and cache the `lineWidth` value (and check for `lineWidth === null` or `undefined` using `!lineWidth`):

```javascript
// ...
var errMargin = 1.01;
var lineWidth;

//...

lineWidth = helpers.valueOrDefault(vm.borderWidth, defaults.global.elements.point.borderWidth);
ctx.strokeStyle = !lineWidth ? 'rgba(0,0,0,0)' : vm.borderColor || defaultColor;
ctx.lineWidth = lineWidth;
```